src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/GetAzureNetworkWatcherConnectionMonitorReportCommand.cs

```diff
@@ -0,0 +1,151 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using AutoMapper;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Network;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmNetworkWatcherConnectionMonitorReport", DefaultParameterSetName = "SetByName"), OutputType(typeof(PSConnectionMonitorQueryResult))]
```

Comment:
@irrogozh this was mentioned before in an email from @mattreatMSFT , but we should look into making this functionality a part of the object returned from `Get-AzureRmNetworkWatcherConnectionMonitor`. If the user wants to get this information, then they would provide a `ShowReport` or `Detailed` switch parameter, which would make the extra API call to get this information and add it as a property of the returned object. Let me know what you think. Any time we can reduce the number of cmdlets we have users looking through, we should look into it.

Reply:
Cormac, I discussed this with team several times. We decided to keep both cmdlets.